pkg/controller.v1/pytorch/env.go

```diff
@@ -0,0 +1,165 @@
+// Copyright 2021 The Kubeflow Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pytorch
+
+import (
+	"errors"
+	"strconv"
+	"strings"
+
+	"github.com/kubeflow/tf-operator/pkg/common/jobcontroller"
+	corev1 "k8s.io/api/core/v1"
+
+	v1 "github.com/kubeflow/pytorch-operator/pkg/apis/pytorch/v1"
+)
+
+const (
+	EnvMasterAddr       = "MASTER_ADDR"
+	EnvMasterPort       = "MASTER_PORT"
+	EnvWorldSize        = "WORLD_SIZE"
+	EnvRank             = "RANK"
+	EnvPythonUnbuffered = "PYTHONUNBUFFERED"
+
+	// EnvPetMasterAddr is the environment variable name for the master address.
+	EnvPetMasterAddr = "PET_MASTER_ADDR"
+	// EnvPetMasterPort is the environment variable name for the master port.
+	EnvPetMasterPort = "PET_MASTER_PORT"
+	// EnvPetNprocPerNode is the environment variable name for the number of processes per node.
+	EnvPetNprocPerNode = "PET_NPROC_PER_NODE"
+	// EnvPetNnodes is the environment variable name for the number of nodes.
+	EnvPetNnodes = "PET_NNODES"
+	// EnvPetNodeRank is the environment variable name for the rank of nodes.
+	EnvPetNodeRank = "PET_NODE_RANK"
+)
+
+// setPodTemplateEnv sets the environment variables for the podTemplateSpec.
+func setPodTemplateEnv(podTemplateSpec *corev1.PodTemplateSpec, job *v1.PyTorchJob, rtype v1.PyTorchReplicaType, index string) error {
+	rank, err := strconv.Atoi(index)
```

Comment:
I suggest adding  unit test for  the new function setPodTemplateEnv to  confirm that it correctly sets all necessary environment variables.

Reply:
Done.